src/evict/evict_lru.c

```diff
@@ -1559,9 +1582,12 @@ __evict_get_ref(
 		candidates /= 2;
 
 	/* Get the next page queued for eviction. */
-	while ((evict = cache->evict_current) != NULL &&
-	    evict < evict_queue->evict_queue + candidates &&
-	    evict->ref != NULL) {
+	for (evict = cache->evict_current;
```

Comment:
No we can't, but the loop condition checks that `evict` is between the start of the queue and the end, so it will catch `NULL`.